baseplate/frameworks/thrift/__init__.py

```diff
@@ -82,7 +82,10 @@ def _set_remote_context(self, request_context: RequestContext) -> Iterator[None]
             header_dict = {}
             for k, v in headers.items():
                 try:
-                    header_dict[k.decode()] = v.decode()
+                    # this is only for w3c trace headers
+                    key = k.decode()
+                    if key in ["traceparent", "tracestate", "Trace", "Span", "Sampled", "Flags"]:
```

Comment:
can we move this slightly magic array to a set somewhere?

W3C_HEADERS = {...} ?

Reply:
+1, maybe `W3C_HEADERS = frozenset((...))` as a general best practice so it's not mutable.